phac-nml/irida-next
# Attachments: Checksum Uniqueness Validation

## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR introduces Checksum uniqueness validation to the `Attachment` model. This prevents more than one `Attachment` from being associated to an `Attachable` where the `file` computed checksum matches an existing `file` in another `Attachment` associated with the `Attachable`.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

![image](https://github.com/phac-nml/irida-next/assets/492127/d430dbad-059d-4bd8-9859-53e514857df8)

![image](https://github.com/phac-nml/irida-next/assets/492127/d2f6e680-0091-4add-b0c1-a0bcb77bbfb5)

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Upload `test_file.fastq` to a Sample. 
2. Attempt to reupload `test_file.fastq` to the same Sample.
3. See that an error message appears and that the file was not uploaded.

Note: `test_file.fastq` can be found at `test/fixtures/files` in the repository.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
